.github/workflows/continous-integration-docker.yml

```diff
@@ -239,10 +246,13 @@ jobs:
       height_grep_text: 'finished initial sync to chain tip, using gossiped blocks sync_percent=100.* current_height=Height'
 
   # Test that Zebra can answer a synthetic RPC call, using a cached Zebra tip state
+  #
+  # If the state version has changed, waits for the new cached state to be created.
+  # Otherwise, if the state rebuild was skipped, runs immediately after the build job.
   lightwalletd-rpc-test:
-    needs: build
+    needs: test-full-sync
     uses: ./.github/workflows/deploy-gcp-tests.yml
-    if: ${{ github.event.inputs.regenerate-disks != 'true' && github.event.inputs.run-full-sync != 'true' }}
+    if: ${{ !cancelled() && !failure() && github.event.inputs.regenerate-disks != 'true' && github.event.inputs.run-full-sync != 'true' }}
```

Comment:
I'm not sure on the expected behavior from `!cancelled() && !failure()`

Shouldn't be this something like `needs.regenerate-stateful-disks.outputs.conclusion != 'cancelled'` ?

As I'm supposing this is trying to get the previous job conclusion status, and based on that then decide if this job will run or not.

Reply:
> I'm not sure on the expected behavior from `!cancelled() && !failure()`

Here's the behaviour I want:
https://github.com/ZcashFoundation/zebra/blob/b841fe71b2be706f8dcb18a953127bdca85a8773/.github/workflows/continous-integration-docker.yml#L250-L251

But I can't write `success() || skipped()`, because there is no `skipped()`.
So instead I have to write `!cancelled() && !failure()`.

`cancelled()`:
> > Returns true if the workflow was canceled.

`failure()`:
> > Returns true when any previous step of a job fails. If you have a chain of dependent jobs, failure() returns true if any ancestor job fails.

https://docs.github.com/en/actions/learn-github-actions/expressions#cancelled

> Shouldn't be this something like `needs.regenerate-stateful-disks.outputs.conclusion != 'cancelled'` ?

This does the same thing as `cancelled()`.

> As I'm supposing this is trying to get the previous job conclusion status, and based on that then decide if this job will run or not.

Yes, and that works, but the disk rebuild condition doesn't:
https://github.com/ZcashFoundation/zebra/pull/4385#issuecomment-1125623757

Reply:
Conditionals reference: https://github.com/actions/runner/issues/491#issuecomment-926924523